js/fileinput.js

```diff
@@ -1261,17 +1487,32 @@
                     self._ajaxSubmit(fnBefore, fnSuccess, fnComplete, fnError, fd, id, rm.fileIndex, opts.testUrl);
                 },
                 pushAjax: function (index, retry) {
+                    var tm = self.taskManager, rm = self.resumableManager, tasksPool = tm.getTasksPool(rm.id);
+
+                    tasksPool.addTask(tasksPool.getAllTasksCount() + 1, function(deferrer) {
+
+                        // use fifo chunk stack
+                        var arr = rm.stack.shift(), index;
+
+                        index = arr[0];
+                        if (!rm.chunksProcessed[rm.id] || !rm.chunksProcessed[rm.id][index]) {
+                            rm.sendAjax(index, arr[1], deferrer);
+                            return;
+                        } else {
+                            console.log('wtf');
```

Comment:
There is one issue I am seeing on the pause and resume. When you pause it once and then resume it works fine. But when you pause and resume more than once for the same file - the code runs into an issue and enters this console.log... need to check.

Reply:
OKay i'll look at that.